pkg/policy/l4.go

```diff
@@ -231,10 +231,10 @@ func (l7 L7DataMap) addRulesForEndpoints(rules api.L7Rules, endpoints []CachedSe
 
 // delete releases the references held in the L4Filter and must be called before
 // the filter is left to be garbage collected!
-func (l4 *L4Filter) delete(selectorCache *SelectorCache) {
+func (l4 *L4Filter) detach(selectorCache *SelectorCache) {
 	selectorCache.RemoveSelectors(l4, l4.CachedSelectors)
-	l4.CachedSelectors = nil
-	l4.L7RulesPerEp = nil
+	//l4.CachedSelectors = nil
```

Comment:
Why do these no longer need to be set to nil?

Reply:
The L4 policy may still be used (temporarily) as a realized or desired policy of an endpoint, until the endpoint completes regeneration. So we must not change the L4 policy at all, but we still need the remove the L4 Policy from the selector cache users.

This is why I renamed this to "Detach()" (from "Delete()").